machinist/mnode/mnode.go

```diff
@@ -0,0 +1,69 @@
+package mnode
+
+import (
+	"context"
+	"fmt"
+	"github.com/enfabrica/enkit/machinist"
+	machinist_rpc "github.com/enfabrica/enkit/machinist/rpc/machinist"
+	"google.golang.org/grpc"
+	"time"
+)
+
+type Node struct {
+	Client machinist_rpc.ControllerClient
+	*machinist.SharedFlags
+	Name string
+	Tags []string
+	// Dial func will override any existing options to connect
+	DialFunc func() (*grpc.ClientConn, error)
+}
+
+func (n *Node) Flags() *machinist.SharedFlags {
+	return n.SharedFlags
+}
+
+func (n *Node) Init() error {
+	if n.DialFunc != nil {
```

Comment:
nit: would invert the if to reduce nesting. Given this still needs to be implemented, I'm ok merging as is, and addressing later.